src/packet.c

```diff
@@ -0,0 +1,72 @@
+#include "packet.h"
+
+
+Packet to_packet(struct sk_buff *skb) {
+    Packet packet = {};
+
+    if (!skb) {
+        return packet;
+    }
+
+    packet.ip_header = ip_hdr(skb);
+    if (packet.ip_header) {
+        switch (packet.ip_header->protocol)
+        {
+            case IPPROTO_TCP:
```

Comment:
Since you support only TCP packets in this section, why proceed with other protocols and not continue to next packet?